exchanges/gateio/gateio_wrapper.go

```diff
@@ -2172,21 +2173,110 @@ func (g *Gateio) UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) e
 				minBaseAmount = math.Pow10(-int(pairsData[x].AmountPrecision))
 			}
 
-			limits = append(limits, order.MinMaxLevel{
-				Asset:                   a,
-				Pair:                    pair,
+			l = append(l, limits.MinMaxLevel{
+				Key:                     key.NewExchangePairAssetKey(g.Name, a, pair),
 				QuoteStepIncrementSize:  math.Pow10(-int(pairsData[x].Precision)),
 				AmountStepIncrementSize: math.Pow10(-int(pairsData[x].AmountPrecision)),
 				MinimumBaseAmount:       minBaseAmount,
 				MinimumQuoteAmount:      pairsData[x].MinQuoteAmount.Float64(),
 			})
 		}
+	case asset.Futures:
+		btcContracts, err := g.GetAllFutureContracts(ctx, currency.BTC)
+		if err != nil {
+			return err
+		}
+		usdtContracts, err := g.GetAllFutureContracts(ctx, currency.USDT)
+		if err != nil {
+			return err
+		}
+		btcContracts = append(btcContracts, usdtContracts...)
+		l = make([]limits.MinMaxLevel, 0, len(btcContracts))
+		for x := range btcContracts {
+			if btcContracts[x].InDelisting {
+				continue
+			}
```

Comment:
Probably don't need this and below, if its loaded in memory anyway its going to persist. If you have long or short contract open on startup you can at least reduce only it with the correct trading details.

Reply:
Do you find that the order execution limits change when the currency is changed to delisting?

Reply:
No, I haven't experienced that when it goes into delisting for the old g i o

Reply:
I still rm'd for you 🌈